src/NuGet.Clients/NuGet.PackageManagement.UI/Xamls/SolutionView.xaml.cs

```diff
@@ -172,13 +175,38 @@ private void SortByColumn(GridViewColumnHeader sortColumn)
             {
                 if (column == sortColumn)
                 {
+                    UpdateHeaderAutomationProperties(column);
                     continue;
                 }
 
                 SortableColumnHeaderAttachedProperties.RemoveSortDirectionProperty(obj: column);
+                UpdateHeaderAutomationProperties(column);
             }
         }
 
+        private void UpdateHeaderAutomationProperties(GridViewColumnHeader columnHeader)
+        {
+            var sortDir = SortableColumnHeaderAttachedProperties.GetSortDirectionProperty(columnHeader);
+            string oldHelpText = AutomationProperties.GetHelpText(columnHeader);
+            string newHelpText;
+            if (sortDir == ListSortDirection.Ascending)
+            {
+                newHelpText = Resx.Resources.Accessibility_ColumnSortedAscendingHelpText;
+            }
+            else if (sortDir == ListSortDirection.Descending)
+            {
+                newHelpText = Resx.Resources.Accessibility_ColumnSortedDescendingHelpText;
+            }
+            else
+            {
+                newHelpText = Resx.Resources.Accessibility_ColumnNotSortedHelpText;
+            }
+
+            AutomationProperties.SetHelpText(columnHeader, newHelpText);
+            var peer = UIElementAutomationPeer.FromElement(columnHeader);
```

Comment:
Can we move this null check to the beginning. No reason to parse values/strings if there's no peer, right?

Reply:
@donnie-msft The null check for the peer is to make sure it's not null when we try to raise the event that the HelpText changed so the screen reader re-reads it but I think we'd want to set the HelpText even if the peer is null. That's why I'm doing the null check just for raising the property changed event but do you think the HelpText should not be set at all if the automation peer were null?

Reply:
I haven't tested it, but based on my knowledge and https://docs.microsoft.com/en-us/windows/winui/api/microsoft.ui.xaml.automation.automationproperties?view=winui-3.0-preview I thought a null peer would mean the properties are useless. What does an AutomationProperty do when there's no peer?

Reply:
I know that an AutomationProperty attached property can still get a value when there is no AutomationPeer so my thought was that it was good to set the property so if / when there is an AutomationPeer, it will read the HelpText when the column header gets focused. So I only check for null when raising the event that forces it to re-read the HelpText but set it regardless.

Reply:
In other words, I have the same understanding that the AutomationProperties won't do anything if there is no AutomationPeer, what I am not sure of is whether or not the automation peer can be created later or exactly when it gets created in the chain of events. If it can get created after the properties are set, then it's good to set the properties for when the peer exists but only raise events on the peer if it exists at that time.

Reply:
It is common practice to set AutomationProperties without checking for an AutomationPeer, as is done in XAML, and the code to set the HelpText is inexpensive so I think the best bet is to not optimize this.